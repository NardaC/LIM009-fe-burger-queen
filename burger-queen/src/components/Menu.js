import React, { useState } from 'react';
import Desayuno from './Desayuno';
import Almuerzo from './Almuerzo';
import Acompañamientos from './Acompa\u00F1amientos';
import Bebidas from './Bebidas';

const Menu = () => {
    const [tipo, setTipo] = useState('desayuno')
    return(
        <>
            <div>
                <button onClick={() => {setTipo('desayuno')}} type="button" className="btn btn-info">Desayuno</button>
                <button onClick={() => {setTipo('almuerzo')}} type="button" className="btn btn-info">Almuerzo y Cena</button>
                <button onClick={() => {setTipo('acompañamiento')}} type="button" className="btn btn-info">Acompañamientos</button>
                <button onClick={() => {setTipo('bebidas')}} type="button" className="btn btn-info">Bebidas</button>      
            </div>
            <div data-testid='content'>
                {tipo === 'desayuno' && (
                    <div data-testid='desayuno'>
                         <Desayuno/>
                    </div>
                )}
                {tipo === 'almuerzo' && (
                    <div>
                        <Almuerzo/>
                    </div>
                )}
                {tipo === 'acompañamiento' && (
                    <div>
                         <Acompañamientos/>
                    </div>
                )}
                {tipo === 'bebidas' && (
                    <div>
                         <Bebidas/>
                    </div>
                )}
            </div>
        </>
    )
    
    }
 export default Menu;