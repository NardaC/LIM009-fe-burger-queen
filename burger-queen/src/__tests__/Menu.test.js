import React from "react";
import { render, fireEvent, waitForElement, act } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@testing-library/react/cleanup-after-each";
import "@testing-library/jest-dom/extend-expect";
import Menu from "../components/Menu";

it("renders", () => {
  const { asFragment } = render(<Menu />);
  expect(asFragment()).toMatchSnapshot();
});
it("inserts", async () => {
  const { getByTestId, getByText } = render(<Menu />);
const btn = getByText("Desayuno")
  act(() => {
    fireEvent.click(btn);
  });
  const desayunoElement = await waitForElement(() => {
    return getByTestId("desayuno");
  });
  // expect(getByTestId('imgLogo')).toHaveClass('center');
  // expect(getByAltText('logo')).toHaveAttribute('alt', expect.stringContaining('logo'));
  expect(desayunoElement.textContent).toBe("Collection: Loading...");
});
