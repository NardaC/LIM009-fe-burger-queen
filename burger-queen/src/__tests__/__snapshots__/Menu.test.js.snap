// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn btn-info"
      type="button"
    >
      Desayuno
    </button>
    <button
      class="btn btn-info"
      type="button"
    >
      Almuerzo y Cena
    </button>
    <button
      class="btn btn-info"
      type="button"
    >
      Acompa√±amientos
    </button>
    <button
      class="btn btn-info"
      type="button"
    >
      Bebidas
    </button>
  </div>
  <div
    data-testid="content"
  >
    <div
      data-testid="desayuno"
    >
      <div>
        <span>
          Collection: Loading...
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
